tools/calyx-ffi-macro/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+use std::{env, path::PathBuf};
+
+use parse::{CalyxFFIMacroArgs, CalyxPortDeclaration};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, spanned::Spanned};
+
+mod calyx;
+mod parse;
+mod util;
+
+#[proc_macro_attribute]
+pub fn calyx_ffi(attrs: TokenStream, item: TokenStream) -> TokenStream {
+    let source_manifest_dir = PathBuf::from(
+        env::vars()
+            .find(|(name, _)| name == "CARGO_MANIFEST_DIR")
+            .expect("caller of calyx_ffi did not use cargo to build project")
+            .1,
+    );
+
+    let args = parse_macro_input!(attrs as CalyxFFIMacroArgs);
+    let item_struct = parse_macro_input!(item as syn::ItemStruct);
+    let name = item_struct.ident;
+    let given_path = args.src.to_string_lossy().to_string();
+
+    let mut path = source_manifest_dir;
+    path.push(given_path);
+    let path = path;
+
+    // <sorry>
```

Comment:
Please replace with a more descriptive comment. Why might this break in the future? What is the problem with this code?

Reply:
It's just silly code. I can remove the comments.